core/src/test/java/org/apache/calcite/sql/type/SqlTypeUtilTest.java

```diff
@@ -117,6 +131,57 @@
     SqlTypeCoercionRule.THREAD_PROVIDERS.set(defaultRules);
   }
 
+  @Test void testConvertTypeToSpec() {
+    final SqlTypeFactoryImpl factory =
+        new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+    final RelDataType bigint = factory.createSqlType(SqlTypeName.BIGINT);
```

Comment:
Try to reuse the type defined in `f`,if there is no what you need, create one there.

Reply:
Thanks for your advice.